lib/bsm.js

```diff
@@ -2,149 +2,134 @@
-      message.address = address
+    return message.verify().verified
+  }
 
-      return message.verify().verified
-    }
+  static async asyncVerify (messageBuf, sigstr, address) {
+    let args = [messageBuf, sigstr, address]
+    let workersResult = await Workers.asyncClassMethod(Bsm, 'verify', args)
+    let res = JSON.parse(workersResult.resbuf.toString())
+    return res
+  }
 
-    static async asyncVerify (messageBuf, sigstr, address) {
-      let args = [messageBuf, sigstr, address]
-      let workersResult = await Workers.asyncClassMethod(Bsm, 'verify', args)
-      let res = JSON.parse(workersResult.resbuf.toString())
-      return res
-    }
+  sign () {
+    let hashBuf = Bsm.magicHash(this.messageBuf)
+    let ecdsa = new Ecdsa().fromObject({
+      hashBuf: hashBuf,
+      keyPair: this.keyPair
+    })
+    ecdsa.sign()
+    ecdsa.calcrecovery()
+    this.sig = ecdsa.sig
+    return this
+  }
 
-    sign () {
-      let hashBuf = Bsm.magicHash(this.messageBuf)
-      let ecdsa = new Ecdsa().fromObject({
-        hashBuf: hashBuf,
-        keyPair: this.keyPair
-      })
-      ecdsa.sign()
-      ecdsa.calcrecovery()
-      this.sig = ecdsa.sig
-      return this
-    }
+  verify () {
+    let hashBuf = Bsm.magicHash(this.messageBuf)
 
-    verify () {
-      let hashBuf = Bsm.magicHash(this.messageBuf)
-
-      let ecdsa = new Ecdsa()
-      ecdsa.hashBuf = hashBuf
-      ecdsa.sig = this.sig
-      ecdsa.keyPair = new KeyPair()
-      ecdsa.keyPair.pubKey = ecdsa.sig2PubKey()
-
-      if (!ecdsa.verify()) {
-        this.verified = false
-        return this
-      }
-
-      let address = new Address().fromPubKey(
-        ecdsa.keyPair.pubKey,
-        undefined,
-        this.sig.compressed
-      )
-      // TODO: what if livenet/testnet mismatch?
-      if (cmp(address.hashBuf, this.address.hashBuf)) {
-        this.verified = true
-      } else {
-        this.verified = false
-      }
+    let ecdsa = new Ecdsa()
+    ecdsa.hashBuf = hashBuf
+    ecdsa.sig = this.sig
+    ecdsa.keyPair = new KeyPair()
+    ecdsa.keyPair.pubKey = ecdsa.sig2PubKey()
 
+    if (!ecdsa.verify()) {
+      this.verified = false
       return this
     }
-  }
 
-  Bsm.magicBytes = Buffer.from('Bitcoin Signed Message:\n')
+    let address = new Address().fromPubKey(
+      ecdsa.keyPair.pubKey,
+      undefined,
+      this.sig.compressed
+    )
+    // TODO: what if livenet/testnet mismatch?
+    if (cmp(address.hashBuf, this.address.hashBuf)) {
+      this.verified = true
+    } else {
+      this.verified = false
+    }
 
-  return Bsm
+    return this
+  }
 }
 
-inject = require('./injecter')(inject, dependencies)
-let Bsm = inject()
-Bsm.Mainnet = inject({
-  KeyPair: require('./key-pair').Mainnet
-})
-Bsm.Testnet = inject({
-  KeyPair: require('./key-pair').Testnet
-})
+Bsm.magicBytes = Buffer.from('Bitcoin Signed Message:\n')
+
+// TODO: Discuss whether this needs to be supported, it is not used anywhere
```

Comment:
It seems fine to me to get rid of this. We don't need to propagate complexity around testnet everywhere. If someone actually needs testnet BSM, we can add it back in.